Scripts/HW_OOP.py

```diff
@@ -0,0 +1,483 @@
+# 
+# 
+# Маленькое замечание: По-хорошему, между классом `BiologicalSequence` и классами `NucleicAcidSequence` и `AminoAcidSequence`, ещё должен быть класс-прослойка, частично реализующий интерфейс `BiologicalSequence`, но его писать не обязательно, так как задание и так довольно большое (правда из-за этого у вас неминуемо возникнет повторяющийся код в классах `NucleicAcidSequence` и `AminoAcidSequence`)
+
+
+
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+    
+    @abstractmethod
+    def __len__():
+        pass
+    
+    @abstractmethod
+    def __getitem__():
+        pass
+    
+    @abstractmethod
+    def __repr__():
+        pass
+    
+    @abstractmethod
+    def check_alphabet():
+        pass
+    
+    
+    def __getattribute__(self, attribute):
+        if "forbiden_methods" in attribute:
+            try:
+                return super().__getattribute__(attribute)
+            except AttributeError:
+                return []
+        forbiden_methods = self._forbiden_methods
+        for method in forbiden_methods:
+            if method in attribute:
+                raise NotImplementedError(f"Method {attribute}() is forbiden")
+        return super().__getattribute__(attribute)
+
+
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    
+    def __init__(self, sequence):
+        self.sequence = sequence
+        self._forbiden_methods = ["complement", "check_alphabet"]
+    
+    
+    def __len__(self):
+        return len(self.sequence)
+    
+    
+    def __getitem__(self, slc):
+        return self.sequence[slc]
+    
+    
+    def __repr__(self):
+        return self.sequence
+    
+    
+    def check_alphabet(self):
+        self.sequence = self.sequence.upper()
+        true_seq = 'AGTC'
+        if set(self.sequence) <= set(true_seq):
+            print('It is okey!')
+        else:
+            raise AlphabetError(f'{self.sequence} is not a true biological nucleotide sequence!')
+    
+    def complement(self):
+        self.sequence = self.sequence.upper()
+        complement_dict = {'A':'T', 'T':'A', 'C':'G', 'G':'C'}
+        seq_complemented = ''
+        for i in self.sequence:
+            seq_complemented += str(complement_dict.get(i))
+        return seq_complemented
+    
+    def gc_content(self):
+        self.sequence = self.sequence.upper()
+        len_seq = len(self.sequence)
+        count_gc = self.sequence.count('G') + self.sequence.count('C')
+        result_gc = round(count_gc/len_seq, 3)
+        return result_gc
+
+
+
+
+
+class DNASequence(NucleicAcidSequence):
+    
+    def __init__(self, sequence):
+        self.sequence = sequence
+    
+    def transcribe(self):
+        self.sequence = self.sequence.upper()
+        transcribe_dict = {'A':'A', 'T':'U', 'C':'C', 'G':'G'}
+        seq_transcribed = ''
+        for i in self.sequence:
+            seq_transcribed += str(transcribe_dict.get(i))
+        print(seq_transcribed)
```

Comment:
Почему `print`? Мы же скорее всего захотим потом как-то использовать полученную последовательность
```suggestion
        return seq_transcribed
```